tests/Propel/Tests/Issues/Issue1598Test.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Propel\Tests\Issues;
+
+use Propel\Generator\Util\QuickBuilder;
+use Propel\Runtime\ActiveQuery\Criteria;
+use Propel\Runtime\ActiveQuery\ModelCriteria;
+use Propel\Runtime\Adapter\Pdo\MysqlAdapter;
+use Propel\Runtime\Propel;
+use Propel\Tests\TestCase;
+
+/**
+ * This test proves the bug described in https://github.com/propelorm/Propel2/issues/1192.
+ *
+ * @link https://github.com/propelorm/Propel2/issues/1192
+ */
+class Issue1192Test extends TestCase
+{
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        if (class_exists('\Issue1192Item')) {
+            return;
+        }
+
+        $schema = <<<END
+<database name="issue_1192">
+    <table name="issue_1192_item">
+        <column name="target" type="INTEGER" required="true" />
+    </table>
+</database>
+END;
+        $builder = new QuickBuilder();
+        $builder->setSchema($schema);
+        $builder->buildClasses(null, true);
+        Propel::getServiceContainer()->setAdapter('issue_1192', new MysqlAdapter());
+    }
+
+    /**
+     * Verifies that the correct SQL and Params are generated for queries that use `Criteria::BINARY_ALL`.
+     */
+    public function testCriteriaBinaryAll()
+    {
+        /** @var ModelCriteria $query */
+        $query = \Issue1192ItemQuery::create()->filterByTarget(1, Criteria::BINARY_ALL);
+        $params = [];
+        $actualSql = $query->createSelectSql($params);
+
+        $this->assertEquals("SELECT  FROM issue_1192_item WHERE issue_1192_item.target & :p1 = :p2",
+            $actualSql,
+            'Generated SQL does not match expected SQL'
+        );
+
+        $this->assertEquals(2, count($params), 'Incorrect number of params');
+
+        $this->assertEquals($params[0], $params[1], 'Params are not identical');
```

Comment:
Can we use assertSame() in those cases? As equals can have false positives with types.